### PR TITLE
Allow compat kind and version selection thru command line and environment var

### PR DESCRIPTION
**Pull request by curtisman**
- Refactor the back compat code into a separate package `test-version-utils`. 
  - `versionUtils.ts` contains code that manages legacy version install, checking and loading,
  - `testApi.ts` contains function that return versioned loader/container runtime/data runtime API.
  - `compatUtils.ts` contains code to generate versioned `TestDataObject`, and versioned `TestObjectProvider`.
  - `compatConfig.ts` contains config and mocha hooks to setup the test variants, controled by command line/environment (using nconf)

Also:
- Switch test to use a more natural `describeFullCompat` instead of `generateTest`.
- Add loader compat only variant via `describeLoaderCompat` and enable loader back compat test variant for `webflow` and `table-document` packages.
- Convert `webflow` test structure to align with the rest of the repo, with separate test build and not us ts-node
- Clean up more direct dependency to `server-local-server`
- Better legacy package install tracking. To detect aborted installed, writing the version in `installed.json` at the end of install as a barrier to indicate that the legacy version is available, instead of just relying on existence of the directory,

(NOTE: this PR includes changes in #5512 but separate out to make it easier to code review)

TODO: Add ability to load different version of `test-driver` thru `test-version-utils` .

### CODE REVIEW

File: examples/data-objects/table-document/package.json [hunk 1]
```diff
@@ -65,9 +65,10 @@
     "@fluidframework/local-driver": "^0.37.0",
     "@fluidframework/mocha-test-setup": "^0.37.0",
     "@fluidframework/runtime-utils": "^0.37.0",
-    "@fluidframework/server-local-server": "^0.1021.0-0",
+    "@fluidframework/test-drivers": "^0.37.0",
```
Comment (anthony-murphy) [inline examples/data-objects/table-document/package.json:68]: >fluidframework [](start = 6, length = 14)

i don't like that we need this dep. hoping with your dynamic loading we can get rid of it :)

File: examples/data-objects/webflow/src/test/.mocharc.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
```
Comment (anthony-murphy) [inline examples/data-objects/webflow/src/test/.mocharc.js:4]: no need to do it in this PR, but we probably want this shareable

**Comment by anthony-murphy** [comment]
> export const debug = registerDebug("fluid:test-e2e");

nit. string is wrong

---
Refers to: packages/test/test-version-utils/src/debug.ts:9 in 261d4d2. [](commit_id = 261d4d261f1f2589b8757d48a9bfcb640727cdca, deletion_comment = False)

**Review by anthony-murphy — APPROVED** [review]
:shipit:

File: examples/data-objects/webflow/src/test/.mocharc.js (hunk 2 above)
Comment (curtisman) [reply examples/data-objects/webflow/src/test/.mocharc.js:4]: Yep... that's the plan.

**Pull request merged**